### DISCUSSION TITLE
Why import ui from '@oplayer/ui' could not Found in Next.js App

### DISCUSSION THREAD

**Discussion by heirro** [shiyiya/oplayer#140] [Q&A] [answered]
Hi Dev,

I try it developing in Next.js App, but i got issue about in when import module `@oplayer/ui`.
Can you explain, or give some examples. By the way, i interested for your player, maybe will build web/mobile app using this Player, since the UI and look fresh and modern XD.

![Screenshot_20241018](https://github.com/user-attachments/assets/78b687c8-8173-4351-b1ce-ea933df889d0)

Thank you!

**Comment by shiyiya** [comment]
![image](https://github.com/user-attachments/assets/504fb788-e507-4cad-9c90-3a5a33f54c1b)

**Reply by heirro** [reply]
Hmm i don't understand, since i just following our Documentation add package like `pnpm add @oplayer/ui`.

Can explain or give example?

![Screenshot_20241018_1](https://github.com/user-attachments/assets/b48703da-4a4a-4946-93e7-cff0870652a3)

**Comment by shiyiya** [comment]
Please provide a minimal repository on GitHub.

**Reply by heirro** [reply]
https://github.com/heirro/oplayer-next

This my example app using next.js and live m3u8

**Comment by shiyiya** [comment]
source type is Source

**Reply by heirro** [reply]
Then i didn't show the video.

```
'use client'

import Player from '@oplayer/react'
import ui from '@oplayer/ui'
import hls from '@oplayer/hls'
import { useEffect, useState, useRef } from 'react'

const plugins = [
  ui({
    pictureInPicture: true,
    slideToSeek: 'always',
    screenshot: true,
    keyboard: { global: true }
  }),
  hls()
]

export default function Home() {
  const [sources, setSources] = useState<string | null>(null);
  const player = useRef<Player>(null);
 
  useEffect(() => {
    // Fetch the source URL for the player
    fetch(`master.m3u8`)
      .then((response) => {
        if (response.ok) {
          return response.url;
        }
        throw new Error('Failed to fetch source');
      })
      .then((url) => {
        setSources(url);
      })
      .catch((error) => {
        console.error('Error fetching source:', error);
      });
  }, []);

  return (
    <>
    <Player
        plugins={plugins}
        ref={player}
        autoplay={true}
        Source={sources}
      />
    </>
  );
}

```

![Screenshot_20241018_2](https://github.com/user-attachments/assets/7f151d56-669e-4d7a-b89b-5f400fda4fc2)

**Comment by shiyiya** [comment, marked as answer]
1. UI TypeError fixed in @oplayer/ui@1.3.4-beta.1
2. React TypeError fixed in @oplayer/react@1.2.11-beta.0
3. **Disable react strict mode**

```
pnpm i @oplayer/ui@1.3.4-beta.1 @oplayer/react@1.2.11-beta.0
```

```js
'use client'

import ReactOPlayer from '@oplayer/react'
import ui from '@oplayer/ui'
import Player, { Source } from '@oplayer/core'

import hls from '@oplayer/hls'
import { useEffect, useState, useRef } from 'react'

const plugins = [
  ui({
    theme: {
      controller: {
        slideToSeek: 'always'
      }
    },
    pictureInPicture: false,
    screenshot: false,
    keyboard: { global: true }
  }),
  hls()
]

export default function Home() {
  const [source, setSource] = useState<Promise<Source>>()
  const player = useRef<Player>(null)

  useEffect(() => {
    setSource(
      fetch('111').then(() => ({
        src: 'https://streamweb.qdei.co:8443/ae96f3d3-e229-4b29-b9f3-68973140b935/master.m3u8'
      }))
    )
  }, [])

  return (
    <>
      <ReactOPlayer plugins={plugins} ref={player} autoplay={false} source={source} />
    </>
  )
}

```

@heirro

**Reply by heirro** [reply]
Thank You!
But i got issue about it.

![Screenshot_20241018_3](https://github.com/user-attachments/assets/963f55fc-85c4-401a-a3bb-8cdfce4e498f)

**Reply by shiyiya** [reply]
check no.3

**Reply by heirro** [reply]
Oh no, i still encounter issues with different things. u,u

![Screenshot_20241018_4](https://github.com/user-attachments/assets/73e36645-9079-40f1-9d42-e3fced436eb4)

**Reply by shiyiya** [reply]
code？

**Reply by heirro** [reply]
Same as the example you provided.

![Screenshot_20241018_5](https://github.com/user-attachments/assets/371110e3-4893-4fcf-ad80-de2272492c43)

**Reply by shiyiya** [reply]
pnpm i @oplayer/react@1.2.11-beta.1

**Reply by heirro** [reply]
BIG THANKS! ❤️‍🔥❤️‍🔥❤️‍🔥
